src/openforms/submissions/models.py

```diff
@@ -146,11 +158,29 @@ class Submission(models.Model):
         ),
     )
 
+    # tracking async execution state
+    on_completion_task_id = models.CharField(
+        _("on completion task ID"),
+        max_length=255,
+        blank=True,
+        help_text=_(
+            "Celery task ID of the on_completion workflow. Use this to inspect the "
+            "state of the async jobs."
+        ),
+    )
+
     objects = SubmissionQuerySet.as_manager()
 
     class Meta:
         verbose_name = _("Submission")
         verbose_name_plural = _("Submissions")
+        constraints = [
+            models.UniqueConstraint(
+                fields=("public_registration_reference",),
+                name="unique_public_registration_reference",
+                condition=~models.Q(public_registration_reference=""),
+            )
```

Comment:
This conditional unique constraint allows us to enforce uniqueness without needing nullable columns, as empty reference values are excluded from the constraint. Also works well with existing data, since it's a new model field.